packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -1060,4 +1065,33 @@ class TextPainter {
     assert(_debugAssertTextLayoutIsValid);
     return  _lineMetricsCache ??= _paragraph!.computeLineMetrics();
   }
+
+  bool _disposed = false;
+
+  /// Whether this object has been disposed or not.
+  ///
+  /// Only for use when asserts are enabled.
+  bool get debugDisposed {
+    bool disposed = false;
```

Comment:
Maybe throw instead of returning `false` in non-debug modes?

Reply:
Done. I think we should come up with a unified strategy for this and put it in the style guide. I'm somewhat inclined to think throwing is too aggressive, but we can hash that out separately from this PR.